detox/src/devices/common/drivers/android/tools/FileXfer.js

```diff
@@ -6,6 +6,23 @@ class FileXfer {
     this._dir = destinationDir;
   }
 
+  async getFileHash(binary) {
+    const crypto = require('./CryptoUtils');
+    return await crypto.getMd5(`${this._dir}/${binary}`);
+  }
+
+  async createEmptyFile(deviceId, filename) {
+    await this._adb.createEmptyFile(deviceId, this._dir, filename);
+  }
+
+  async deleteByExtension(deviceId, extension) {
+    await this._adb.deleteByExtension(deviceId, this._dir, extension);
+  }
+
+  async checkFileExists(deviceId, filename) {
+    return await this._adb.checkFileExists(deviceId, this._dir, filename);
+  }
+
```

Comment:
This shouldn't be a part of FileXfer, because it's not under it's scope. You must find a different design that would better organize this.